app/models/assignment/rotation_generator.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class Assignment < ApplicationRecord
+  class RotationGenerator
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :roster_id, :integer
+    attribute :user_ids
+    attribute :starting_user_id, :integer
+    attribute :start_date, :date
+    attribute :end_date, :date
+
+    validates :roster, presence: true
+    validates :user_ids, presence: true, length: { minimum: 1 }
+    validate :includes_start_user
+    validates :start_date, presence: true
+    validates :end_date, presence: true, comparison: { greater_than_or_equal_to: :start_date }
+
+    def roster
```

Comment:
```suggestion
    def user_ids=(value)
      super(value&.map(&:to_i))
    end

    def roster
```

Here we can add a normalization step to the `user_id=` setter method. This means that the model will automatically cast the ids to integers and we won't have to be careful about type conversions (the unsightly `.to_s` calls) when working with this array